src/main/java/io/vertx/core/impl/future/FutureImpl.java

```diff
@@ -127,6 +127,39 @@ public void onFailure(Throwable failure) {
     return this;
   }
 
+  @Override
+  public Future<T> onComplete(Handler<T> successHandler, Handler<Throwable> failureHandler) {
+    Objects.requireNonNull(successHandler, "No null handler accepted");
```

Comment:
I think that prevents onFailure/onSuccess to work

Reply:
Are you sure? Since this is the FutureImpl, I think this does not relate to the onComplete written in Future?
This method isn't being used in Future is it?
(i.e. the onSuccess and onFailure for FutureImpl are separate?)

If that isn't the case I can remove the null checks from here and instead call them in the addListener onSuccess and onFailure overrides?